Content.Server/Administration/PlaytimeManager.cs

```diff
@@ -0,0 +1,180 @@
+﻿using System;
+using System.Collections.Generic;
+using Content.Server.Afk;
+using Content.Server.Database;
+using Content.Server.Players;
+using Robust.Server.Player;
+using Robust.Shared.Enums;
+using Robust.Shared.IoC;
+using Robust.Shared.Timing;
+
+namespace Content.Server.Administration;
+
+/// <summary>
+///     Tracks a player's <see cref="TotalPlaytime"/> and <see cref="LivingPlaytime"/>, updates every <see cref="UpdateInterval"/> seconds and on roundend.
+/// </summary>
+public interface IPlaytimeManager
+{
+    void Update(float deltaTime);
+
+    void Shutdown();
+
+    public void FetchPlaytime(IPlayerSession session, out int livingTime, out int totalTime);
+
+    float SecondsSinceUpdate { get; set; }
+
+    float UpdateInterval { get; set; }
+
+    bool Ticking { get; set; }
+
+    public void TryLoadPlaytime(IPlayerSession session, PlayerData data);
+}
+
+public class PlaytimeManager : IPlaytimeManager
+{
+    [Dependency] private readonly IPlayerManager _playerManager = default!;
+    [Dependency] private readonly IGameTiming _gameTiming = default!;
+    [Dependency] private readonly IServerDbManager _dbManager = default!;
+    [Dependency] private readonly IAfkManager _afkManager = default!;
+
+    /// <summary>
+    ///     Tracks time between updates. See <see cref="UpdateInterval"/>
+    /// </summary>
+    public float SecondsSinceUpdate { get; set; } = 0f;
+
+    /// <summary>
+    ///     Number of seconds between each <see cref="UpdatePlaytimes"/>. Defaults to 10 minutes.
+    /// </summary>
+    public float UpdateInterval { get; set; } = 600f;
+
+    /// <summary>
+    ///     Whether playtime tracking is active
+    /// </summary>
+    public bool Ticking { get; set; } = true;
+
+    public void Update(float deltaTime)
+    {
+        if (!Ticking) return;
+
+        SecondsSinceUpdate += deltaTime;
+
+        if (SecondsSinceUpdate >= UpdateInterval)
+        {
+            UpdatePlaytimes();
+            SecondsSinceUpdate = 0;
+        }
+    }
+
+    /// <summary>
+    ///     Called during round end to ensure playtime isn't lost.
+    /// </summary>
+    public void Shutdown()
+    {
+        var initialUpdateInterval = UpdateInterval;
+
+        // Give it one last update and then pause until the next round begins.
+        Ticking = false;
+        //This Max() should never be necessary but better safe than sorry.
+        UpdateInterval = Math.Max(UpdateInterval -= SecondsSinceUpdate, 0);
```

Comment:
You shouldn't be mutating `UpdateInterval` in the first place but I would still like to point out this accidental `-=`.